include/irr/video/IGPUObjectFromAssetConverter.h

```diff
@@ -408,6 +413,152 @@ auto IGPUObjectFromAssetConverter::create(asset::ICPUSpecializedShader** const _
     return res;
 }
 
+auto IGPUObjectFromAssetConverter::create(asset::ICPUBufferView** const _begin, asset::ICPUBufferView** const _end) -> created_gpu_object_array<asset::ICPUBufferView>
+{
+    const auto assetCount = std::distance(_begin, _end);
+    auto res = core::make_refctd_dynamic_array<created_gpu_object_array<asset::ICPUBufferView> >(assetCount);
+
+    core::vector<asset::ICPUBuffer*> cpuBufs(assetCount, nullptr);
+    for (ptrdiff_t i = 0u; i < assetCount; ++i)
+        cpuBufs[i] = _begin[i]->getUnderlyingBuffer();
+
+    core::vector<size_t> redirs = eliminateDuplicatesAndGenRedirs(cpuBufs);
+    auto gpuBufs = getGPUObjectsFromAssets(cpuBufs.data(), cpuBufs.data()+cpuBufs.size());
+
+    for (ptrdiff_t i = 0u; i < assetCount; ++i)
+    {
+        asset::ICPUBufferView* cpubufview = _begin[i];
+        IGPUOffsetBufferPair* gpubuf = (*gpuBufs)[redirs[i]].get();
+        (*res)[i] = m_driver->createGPUBufferView(gpubuf->getBuffer(), cpubufview->getFormat(), gpubuf->getOffset() + cpubufview->getOffsetInBuffer(), cpubufview->getByteSize());;
+    }
+
+    return res;
+}
+
+auto IGPUObjectFromAssetConverter::create(asset::ICPUDescriptorSetLayout** const _begin, asset::ICPUDescriptorSetLayout** const _end) -> created_gpu_object_array<asset::ICPUDescriptorSetLayout>
+{
+    const auto assetCount = std::distance(_begin, _end);
+    auto res = core::make_refctd_dynamic_array<created_gpu_object_array<asset::ICPUDescriptorSet> >(assetCount);
+
+    core::vector<asset::ICPUSampler*> cpuSamplers;//immutable samplers
+    size_t maxSamplers = 0ull;
+    size_t maxBindingsPerDescSet = 0ull;
+    size_t maxSamplersPerDescSet = 0u;
+    for (auto dsl : core::SRange<asset::ICPUDescriptorSetLayout*>(_begin, _end))
```

Comment:
I'm not understanding the purpose of this loop

Reply:
compute maxes of array sizes so i dont have to reallocate mem

Reply:
ok but what about the immutable samper stuff?

Reply:
resolved https://github.com/buildaworldnet/IrrlichtBAW/pull/358#discussion_r333191149